test/quantization/core/test_docs.py

```diff
@@ -0,0 +1,96 @@
+# Owner(s): ["oncall: quantization"]
+
+import torch
+from torch.testing._internal.common_quantization import (
+    QuantizationTestCase,
+    SingleLayerLinearModel,
+)
+from os.path import exists
+
+
+class TestQuantizationDocs(QuantizationTestCase):
+    r"""
+    The tests in this section import code from the quantization docs and check that
+    they actually run. In cases where objects are undefined in the code snippet, they
+    must be provided in the test.
+    """
+
+    def _get_code(self, filename, first_line, last_line, offset=2, strict=True):
+        r"""
+        This function reads in the code from the docs, note that first and last
+        line refer to the line number (first line of doc is 1), the offset due to
+        python file reading is handled within this function. Most code snippets
+        have a 2 space indentation, for other indentation levels, change offset,
+        strict=True is to check that the line before the first line and the line after
+        the last line are `newlines`, this is to ensure that the addition of a new line
+        in the docs does not shift the code chunk out of the selection window.
+        """
+        if exists(filename):
+            file = open(filename)
+            content = file.readlines()
+            if strict:
+                assert content[first_line - 2] == "\n" and content[last_line] == "\n", (
+                    "The line before and after the code chunk should be a newline."
+                    " If new material was added to {}, please update this test with"
+                    "the new code chunk line numbers, previously the lines were "
+                    "{} to {}".format(filename, first_line, last_line)
+                )
+
+            code_to_test = ""
+            for i in range(first_line - 2, last_line):
+                code_to_test += content[i][offset:]
+            file.close()
+        else:
+            code_to_test = None
+        return code_to_test
+
+
+    def _test_code(self, code, global_inputs=None):
+        r"""
+        This function runs `code` using any vars in `global_inputs`
+        """
+        if code is not None:  # the path doesn't work for some CI runs
+            expr = compile(code, "test", "exec")
+            exec(expr, global_inputs)
```

Comment:
I think there are 4 cases

1) code is found and passes the test
2) code is found and fails the test (throws an error)
3) code is found but look as we expect (throws an error)
4) can't find the file  (we let this pass)

if your question is about case 3, my worry is that 1 person changes the documentation such that the code in the window passes, then another user edits the API that causes an issue with the code outside the window. To avoid this, the first user should see an error and should have to fix the window so that it hits the correct chunk of code.

thoughts?